### PR TITLE
bug: Remove discussion comment and useless bool return

### LINKED ISSUE #1026: Remove merge comment, clarify wording, and drop useless return value
After https://github.com/mozilla-services/autopush/pull/1023, there was some discussion about the second `get_uaid` call in `WebPushRouter.route_notification`. This question was clarified, however the comment was determined to vague about what was happening.

In addition, there's a useless bool that was being returned from `Message.store_message` that could be removed to prevent similar confusion later.

### PR DESCRIPTION
**Pull request by jrconlin**
Closes #1026

### CODE REVIEW

File: autopush/db.py [hunk 1]
```diff
@@ -436,7 +436,7 @@ def save_channels(self, uaid, channels):
 
     @track_provisioned
     def store_message(self, notification):
-        # type: (WebPushNotification) -> bool
+        # type: (WebPushNotification)
```
Comment (pjenvey) [inline autopush/db.py]: mypy wants -> None here (don't ask me why because I do this all the time)

File: autopush/db.py [hunk 2]
```diff
@@ -448,7 +448,6 @@ def store_message(self, notification):
             updateid=notification.update_id
         )
         self.table.put_item(data=item, overwrite=True)
-        return True
```
Comment (pjenvey) [inline autopush/db.py]: now we should stop checking for this result in route_notification too (and remove the silly "result is False" line)

File: autopush/db.py (hunk 1 above)
Comment (jrconlin) [reply autopush/db.py]: Oh, huh, and pycharm noted that. Need to make it a more important warning.

File: autopush/db.py (hunk 2 above)
Comment (jrconlin) [reply autopush/db.py]: AH! I was looking at the wrong .`save_notification` and wondering what you meant. 

And now I'm disturbed that removing that condition and the returned error didn't cause any test to fail or coverage line to be lost. (sigh)

**Pull request merged**